File: php/bubble-sort/collection.php
<?php
class Collection {
    public $array;

    public function __construct ($array){
        $this->array = $array;
    }

    public function sort(){
        $length = count($this->array);
        $temp = 0;
        for($i = 0; $i< $length -1 ; $i++){
            for($j = 0; $j< $length - $i - 1; $j++){
                if ($this->array[$j] > $this->array[$j+1]){
                    $temp = $this->array[$j];
                    $this->array[$j] = $this->array[$j+1];
                    $this->array[$j+1] = $temp;
                }
            }
        }
    }

    public function display(){
        foreach($this->array as $item){
            echo $item, ' ';
        }
        echo "\n";
    }
}

?>

File: ruby/bubble-sort/collection.rb
class Collection
    @array
    attr_accessor :array
    def initialize(array)
        @array = array
    end
    def sort ()
        i = 0
        j = 0
        temp = 0
        while i < @array.length - 1
            while j < @array.length - i - 1
                if @array[j] > @array[j+1]
                    temp = @array[j]
                    @array[j] = @array[j+1]
                    @array[j+1] = temp
                end
                j += 1
            end
            j=0
            i+=1    
        end
    end
end